ARCs/arc-0060.md

```diff
@@ -0,0 +1,506 @@
+
+```tsx
+export type SignDataFunction = {
+    arbData: StdSignData[],
+    metadata: StdSignMetadata[],
+}
+    => Promise<(SignedData | null)[]>;
+```
+
+- `arbData` is a non-empty array of `StdSigData` objects (defined below).
+- `metadata` is a non-empty array of `StdSignMetadata` objects (defined below) that provide additional information on the data being signed.
+
+The `signData` function returns a list of `SignedData` objects or, in case of error, rejects the promise with an error object `SignDataError`.
+
+#### Interface `HDWalletMetadata`
+
+An `HDWalletMetadata` specifies the derivation path parameters to derive the keys of an HD wallet from the seed.
+
+> HD derivation levels originally from  <a href="https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki">BIP-44</a>.
+
+```tsx
+export interface HDWalletMetadata {
+    /**
+    * HD Wallet purpose value. First derivation path level. 
+    * Hardened derivation is used.
+    */
+    purpose: number,
+
+    /**
+    * HD Wallet coin type value. Second derivation path level.
+    * Hardened derivation is used.
+    */
+    coinType: number,
+
+    /**
+    * HD Wallet account number. Third derivation path level.
+    * Hardened derivation is used.
+    */
+    account: number,
+
+    /**
+    * HD Wallet change value. Fourth derivation path level.
+    * Public derivation is used.
+    */
+    change: number,
+
+    /**
+    * HD Wallet address index value. Fifth derivation path level.
+    * Public derivation is used.
+    */
+    addrIdx: number,
+}
+```
+
+- `purpose` **SHOULD** be set to `44’` for blockchain accounts.
+- `coinType` indicate a derivation subtree for a specific cryptocurrency. It **MUST** be set to `283’` for ALGO according to <a href="https://github.com/satoshilabs/slips/blob/master/slip-0044.md">SLIP-44</a> registered coin types.
+
+The apostrophe in the numbering convention indicates that hardened derivation is used.
+
+#### Interface `StdData`
+
+The arbitrary data **MUST** be represented as a canonicalized JSON object in accordance with <a href="https://www.rfc-editor.org/rfc/rfc8785">RFC 8785</a>.
+
+```tsx
+export type StdData = string;
+```
+
+The `StdData` must be validated with respect the JSON schema provided with `metadata` (defined below). It cannot be a valid Algorand transaction therefore **MUST NOT** be prepended with a known domain separator `TX` or `TG`.
+
+> Algorand domain separators can be found in the <a href="https://github.com/algorandfoundation/specs/blob/master/dev/crypto.md#domain-separation">Algorand specs</a> and in the <a href="https://github.com/algorand/go-algorand/blob/master/protocol/hash.go#L21">Go type HashID</a>.
+
+#### Interface `Ed25519Pk`
+
+An `Ed25519Pk` object is a 32-byte public key, point of the ed25519 elliptic curve. It is base64 encoded in accordance with <a href="https://www.ietf.org/rfc/rfc7468.txt">RFC-7468</a>. The key is not transformed into an Algorand address.
```

Comment:
I would use keys in their raw form. 32 bytes, skip the encoding, specially when this adds size.